contracts/token/ERC20/extensions/ERC4626.sol

```diff
@@ -84,10 +84,10 @@ abstract contract ERC4626 is ERC20, IERC4626 {
      * @dev Attempts to fetch the asset decimals. A return value of false indicates that the attempt failed in some way.
      */
     function _tryGetAssetDecimals(IERC20 asset_) private view returns (bool, uint8) {
```

Comment:
Isn't it because the return value may revert at decoding? A `try IERC20Metadata(address(asset_)).decimals() returns (uint8 value)` would've failed if the token returned an `uint256`. Nothing in the ERC20 restricts the return value to uint8